### ISSUE TITLE
Notification badge for (in-progress) Cold Chain breaches

### ISSUE THREAD

**Issue by adamdewey** [openmsupply/mobile#4142]
## Is your feature request related to a problem? Please describe.

The WHO have requested a demo of the Mobile Cold Chain module as a last step in granting Pre-Qualification accreditation.

Currently there is no mechanism to alert the user of a temperature breach unless you are actually inside the Cold Chain module itself. 

I'm pretty sure the first thing WHO will ask is 'How is the Mobile user notified that there is a temperature breach?' 

It would be a great first step if we could add a notification badge to the Cold Chain module so that a user can see that there is an issue that requires their attention from the Mobile homepage. 

## Describe the solution you'd like

A notification badge that is displayed on the Cold Chain button of the Mobile app homepage. 

**Baseline functionality:** 
- Display a badge on the Mobile homepage when a current breach / multiple breaches are currently in progress. 
- Badge should display the number of sensors with current breaches in progress. 
- Badge should disappear (or number should decrease) once a breach is resolved.

- ~~Badge should appear when there is an unacknowledged temperature breach from any sensor.~~
- ~~Badge should disappear once breach has ~~been viewed by user (entering Cold Chain sensor overview screen is enough).~~ been actively acknowledged by the user - pop-up dialog? Or perhaps tapping on the orange triangle in the sensor graph and acknowledging in the breach detail screen?~~
- ~~Badge should appear even if breach has not yet ended.~~ 

**Desirable functionality:** 
- ~~A Badge is also displayed on the relevant sensor(s) within the Cold Chain sensor page. This badge then disappears once the user taps on the sensor to view the graph / breach detail.~~ 
- ~~Badge acknowledgement should be per user. So if user 1 has seen the breach, the badge should disappear for them, but should not disappear for user 2 until user 2 has seen the breach.~~

![image](https://user-images.githubusercontent.com/19318937/122479199-c8b78a00-d01e-11eb-9b7a-ce66cd4203ac.png)
![image](https://user-images.githubusercontent.com/19318937/122479040-9574fb00-d01e-11eb-81c6-1bf33aeb51b5.png)

## Implementation

**A description of how the solution would best be implemented in the code base**

## Describe alternatives you've considered

**A clear and concise description of any alternative solutions or features you've considered.**

## Additional context

**Add any other context or screenshots about the feature request here.**

**Comment by mark-prins** [comment]
I'm not sure about the per user breach notification - nor the clearing of breaches when a user views the sensor. 

Technically possible I guess.. but to me I think better to require the breach to be actively acknowledged. Which would also mean that if a user logs in and acknowledges a breach ( and there is currently no facility to do this in mobile AFAIK ).. then when another users logs in to the same tablet, they won't see the breach. So if the per user viewing is a requirement then it wouldn't meet the need.

**Comment by adamdewey** [comment]
Sure thing - let's leave the per user for a future improvement. Will strike that. 

Actively acknowledging the breach would be the ideal! (I wasn't sure how hard that would be - but definitely preferable to a passive acknowledgment by just entering the screen). Will update the requirements to reflect that :)

**Comment by katherine-sussol** [comment]
Definitely agree some kind of badge would be useful!
I'm thinking it would be good to split into two sub-issues: one to show the badge on the main menu (which is just how many sensors are currently in breach) and another to handle acknowledgement of the breach to make it go away (worst case if the latter is left out, users just need to sort out the offending sensor/fridge? 🙈 )

**Comment by adamdewey** [comment]
Cool, can definitely break down to make easier to handle!

The only thing is I think it's really important to note we need to show a badge for breaches that have come and gone (as well as sensors currently breaching) - otherwise there's a good chance the user will not notice a brief breach / overnight breach that happened while they were sleeping!

So we need to keep a memory of unacknowledged breaches and display a badge based on that. 

Could make separating the issues a bit trickier? What do you think?

**Comment by adamdewey** [comment]
One option would be to split tickets into: 

- Display badge for current breaches
- Display badge for past breaches

**Comment by adamdewey** [comment]
After speaking to Kat, I think we should go for the easy win here and come back for a more complete solution in another issue. 

So this issue will just be to display a badge notification on the Mobile homepage for a current breach in progress. 

(Will update the issue description above to match)

**Comment by joshxg** [comment]
Sorry this issue is not descriptive enough: https://github.com/openmsupply/mobile/issues/3930

**Comment by adamdewey** [comment]
Ooh, yes Josh - nice one, that would be perfect :)

**Comment by katherine-sussol** [comment]
Closing this issue with #4145, I think maybe we can revisit the badge logic once #3930 is done (this way we avoid having un-actionable badges at any given point? 🤔 )

**Comment by ArpitaSussol** [comment]
Tested in V8.3.0

## Testing

Steps to reproduce or otherwise test the changes of this PR:

- [x] Have some sensors with breaches that have happened at some point, navigate to the main menu page, there should be a badge with the total number of unacknowledged breaches
